File: 240427/독서실의 거리두기 4/study-cafe-keeping-distance-4.py
N = int(input())
seats = list(input())

max_dist = 0
for i in range(N):
    if seats[i] == '1':
        continue

    dist = float("inf")
    for j in range(N):
        if seats[j] == '1' or i == j:
            continue
        
        dist = N + 2
        seats[i], seats[j] = '1', '1'
        for k in range(N):
            for l in range(k + 1, N):
                if seats[k] == '1' and seats[l] == '0':
                    dist = min(dist, l - k + 1)

        max_dist = max(max_dist, dist)
        seats[i], seats[j] = '0', '0'

print(max_dist)

File: 260325/독서실의 거리두기 4/study-cafe-keeping-distance-4.cpp
#include <bits/stdc++.h>
using namespace std;

int N;
vector<char> seats;

int get_dist() 
{
    int min_num = 2100000000;
    int last_flag = -1, cur_flag = -1;

    for (int i = 0; i < N; i++) 
    {
        char cur_chr = seats[i];
        if (cur_chr == '1') 
        {
            cur_flag = i;
        } 
        else 
        {
            cur_flag = -1;
        }

        if (cur_flag != -1) 
        {
            if (last_flag != -1) 
            {
                int sub = cur_flag - last_flag;
                min_num = min(min_num, sub);
            }
            last_flag = cur_flag;
        }
    }

    return min_num;
}

int main() 
{
    cin >> N;
    string s;
    cin >> s;
    seats.assign(s.begin(), s.end());

    // Please write your code here.
    int max_dist = 0;

    for (int i = 0; i < N; i++) 
    {
        for (int j = i + 1; j < N; j++) 
        {
            if (seats[i] == '1' || seats[j] == '1') 
            {
                continue;
            }

            seats[i] = '1';
            seats[j] = '1';

            max_dist = max(max_dist, get_dist());

            seats[i] = '0';
            seats[j] = '0';
        }
    }

    cout << max_dist << '\n';
    return 0;
}